### PR TITLE
Add ability to connect/sign with Freighter in transaction builder

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: src/components/TxBuilderAttributes.js [hunk 1]
```diff
@@ -11,11 +11,34 @@ import {StrKey} from 'stellar-sdk';
...
+  )
+}
+
 export default function TxBuilderAttributes(props) {
   let {onUpdate, attributes} = props;
 
   return <div className="TransactionAttributes">
     <div className="TransactionOp__config TransactionOpConfig optionsTable">
+        {window.lyra ? <ConnectWithLyra onUpdate={onUpdate} /> : null}
+
```
Comment (piyalbasu) [inline src/components/TxBuilderAttributes.js:41]: this mimics how a client site would enable "Connect With Lyra" buttons in real life. Unless you have the chrome extension installed, a user shouldn't have `window.lyra` available to them

File: src/components/TxLyraSign.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    } catch (e) {
+      res = e;
+    }
+
+    setTxResult(res.transactionStatus);
+
+}
+
+const TxLyraSign = ({ xdr }) => {
+  const [ txResult, setTxResult ] = React.useState();
```
Comment (piyalbasu) [inline src/components/TxLyraSign.js:19]: take the xdr we've built and sign the transaction with Lyra

File: src/components/TxLyraSign.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+const signWithLyra = async (xdr, setTxResult) => {
+      let res = {transactionStatus: "" };
+
+    try {
+      res = await window.lyra.submitTransaction({
```
Comment (vcarl) [inline src/components/TxLyraSign.js:7]: Marking this as a WIP API, this is proof of concept atm. We should probably talk through possible use cases apps have, naming.

**Comment by acharb** [comment]
@piyalbasu LGTM! 👍

File: src/components/FormComponents/PubKeyPicker.js [hunk 4]
```diff
@@ -1,27 +1,47 @@
...
+            return "Public key is invalid.";
+          }
+        }}
+      />
+      {hasLyra && (
+        <button
+          type="button"
+          onClick={() => {
+            // TODO
+            console.log(new Error("not implemented"));
```
Comment (quietbits) [inline src/components/FormComponents/PubKeyPicker.js:38]: Will this be implemented in later PR?

File: src/components/TransactionSigner.js [hunk 5]
```diff
@@ -171,6 +172,18 @@ class TransactionSigner extends React.Component {
                   >Sign with BIP Path</button>
                   {ledgerwalletMessage}
                 </OptionsTablePair>
+                {isConnected() && <OptionsTablePair label="Lyra">
+                  <button
+                    className="s-button TxSignerKeys__signLyra"
+                    onClick={() => {
+                      // TODO: dispatch(signWithLyra(xdr, networkPassphrase))
+                      console.log(new Error('not implemented'))
```
Comment (quietbits) [inline src/components/TransactionSigner.js:180]: Same question as above.

File: src/components/TxBuilderAttributes.js [hunk 6]
```diff
@@ -14,6 +15,27 @@ import {fetchSequence, fetchBaseFee, updateTxType, updateFeeBumpAttribute} from
...
+  try {
+    response = await getPublicKey();
+  } catch (e) {
+    console.error(e);
+  }
+  const { publicKey, error } = response;
+  onUpdate('sourceAccount', publicKey);
+  }
+
```
Comment (quietbits) [inline src/components/TxBuilderAttributes.js:32]: Indentation and spacing seem off here.

File: src/components/TxBuilderResult.js [hunk 7]
```diff
@@ -48,55 +51,89 @@ class TxBuilderResult extends React.Component {
...
-        xdrLink = <a className="s-button"
-          href={xdrViewer(transactionBuild.xdr, 'TransactionEnvelope')}
-          onClick={scrollOnAnchorOpen}>View in XDR Viewer</a>
+        );
+        signingInstructions = (
+          <p className="TransactionBuilderResult__instructions">
+            In order for the transaction to make it into the ledger, a
+            transaction must be successfully signed and submitted to the
+            network. The laboratory provides the{" "}
+            <a href="#txsigner">Transaction Signer</a> to for signing a
```
Comment (quietbits) [inline src/components/TxBuilderResult.js:87]: Extra "to" here?

**Comment by quietbits** [comment]
Looks good! :+1: Just added a couple of tiny comments that could be easy to forget.

**Review by quietbits — APPROVED** [review]

**Review by vcarl — APPROVED** [review]
🎉

**Pull request merged**